### PR TITLE
Role icon support

### PR DESCRIPTION
**Pull request by conorwastakenwastaken**
# New properties

- Role#icon
- Role#iconURL
- Role#unicodeEmoji

# Misc Updates

- Added abilty to create a new role with an icon/emoji (4b5a8ee)
- Added documentation for editing a role icon/emoji (4b5a8ee)
- Added support for the `guildRoleUpdate` event to display role icon/emoji changes (6645864)
- Added documentation for new `ROLE_ICONS` guild feature (3c45b6c)

### CODE REVIEW

File: lib/gateway/Shard.js [hunk 1]
```diff
@@ -1394,12 +1396,14 @@ class Shard extends EventEmitter {
                 * @prop {Object} oldRole The old role data
                 * @prop {Number} oldRole.color The hex color of the role in base 10
                 * @prop {Boolean} oldRole.hoist Whether users with this role are hoisted in the user list or not
+                * @prop {String?} icon The hash of the role's icon image, or null if no icon
```
Comment (DonovanDMC) [inline lib/gateway/Shard.js]:
```suggestion
                * @prop {String?} oldRole.icon The hash of the role's icon image, or null if no icon
```

File: lib/gateway/Shard.js [hunk 2]
```diff
@@ -1394,12 +1396,14 @@ class Shard extends EventEmitter {
...
                 * @prop {Boolean} oldRole.managed Whether a guild integration manages this role or not
                 * @prop {Boolean} oldRole.mentionable Whether the role is mentionable or not
                 * @prop {String} oldRole.name The name of the role
                 * @prop {Permission} oldRole.permissions The permissions number of the role
                 * @prop {Number} oldRole.position The position of the role
                 * @prop {Object?} oldRole.tags The tags of the role
+                * @prop {String?} unicodeEmoji Unicode emoji for the role
```
Comment (DonovanDMC) [inline lib/gateway/Shard.js]:
```suggestion
                * @prop {String?} oldRole.unicodeEmoji Unicode emoji for the role
```

**Review by DonovanDMC — CHANGES_REQUESTED** [review]

**Comment by bsian03** [comment]
it's not october yet

File: index.d.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ declare namespace Eris {
   // Guild
   type DefaultNotifications = 0 | 1;
   type ExplicitContentFilter = 0 | 1 | 2;
-  type GuildFeatures = "ANIMATED_ICON" | "BANNER" | "COMMERCE" | "COMMUNITY" | "DISCOVERABLE" | "FEATURABLE" | "INVITE_SPLASH" | "MEMBER_VERIFICATION_GATE_ENABLED" | "NEWS" | "PARTNERED" | "PREVIEW_ENABLED" | "VANITY_URL" | "VERIFIED" | "VIP_REGIONS" | "WELCOME_SCREEN_ENABLED" | "TICKETED_EVENTS_ENABLED" | "MONETIZATION_ENABLED" | "MORE_STICKERS" | "THREE_DAY_THREAD_ARCHIVE" | "SEVEN_DAY_THREAD_ARCHIVE" | "PRIVATE_THREADS";
+  type GuildFeatures = "ANIMATED_ICON" | "BANNER" | "COMMERCE" | "COMMUNITY" | "DISCOVERABLE" | "FEATURABLE" | "INVITE_SPLASH" | "MEMBER_VERIFICATION_GATE_ENABLED" | "NEWS" | "PARTNERED" | "PREVIEW_ENABLED" |"ROLE_ICONS" | "VANITY_URL" | "VERIFIED" | "VIP_REGIONS" | "WELCOME_SCREEN_ENABLED" | "TICKETED_EVENTS_ENABLED" | "MONETIZATION_ENABLED" | "MORE_STICKERS" | "THREE_DAY_THREAD_ARCHIVE" | "SEVEN_DAY_THREAD_ARCHIVE" | "PRIVATE_THREADS";
```
Comment (DonovanDMC) [inline index.d.ts]:
```suggestion
  type GuildFeatures = "ANIMATED_ICON" | "BANNER" | "COMMERCE" | "COMMUNITY" | "DISCOVERABLE" | "FEATURABLE" | "INVITE_SPLASH" | "MEMBER_VERIFICATION_GATE_ENABLED" | "NEWS" | "PARTNERED" | "PREVIEW_ENABLED" | "ROLE_ICONS" | "VANITY_URL" | "VERIFIED" | "VIP_REGIONS" | "WELCOME_SCREEN_ENABLED" | "TICKETED_EVENTS_ENABLED" | "MONETIZATION_ENABLED" | "MORE_STICKERS" | "THREE_DAY_THREAD_ARCHIVE" | "SEVEN_DAY_THREAD_ARCHIVE" | "PRIVATE_THREADS";
```

Spacing
(and honestly, we should split this up between lines at some point, this is a mess {maybe by similar being grouped?})

**Review by DonovanDMC — CHANGES_REQUESTED** [review]

File: lib/Client.js [hunk 4]
```diff
@@ -1311,9 +1315,11 @@ class Client extends EventEmitter {
     * @arg {Object} options The properties to edit
     * @arg {Number} [options.color] The hex color of the role, in number form (ex: 0x3da5b3 or 4040115)
     * @arg {Boolean} [options.hoist] Whether to hoist the role in the user list or not
+    * @arg {String} [options.icon] The role icon as a base64 data URI
     * @arg {Boolean} [options.mentionable] Whether the role is mentionable or not
     * @arg {String} [options.name] The name of the role
     * @arg {BigInt | Number | String | Permission} [options.permissions] The role permissions
```
Comment (bsian03) [inline lib/Client.js:1321]: Needs snake_case handling

File: lib/gateway/Shard.js [hunk 5]
```diff
@@ -1394,12 +1396,14 @@ class Shard extends EventEmitter {
                 * @prop {Object} oldRole The old role data
                 * @prop {Number} oldRole.color The hex color of the role in base 10
                 * @prop {Boolean} oldRole.hoist Whether users with this role are hoisted in the user list or not
+                * @prop {String?} oldRole.icon The hash of the role's icon image, or null if no icon
```
Comment (bsian03) [inline lib/gateway/Shard.js]: "role's icon image" sounds weird imo

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Review by bsian03 — APPROVED** [review]

**Pull request merged**